app/routers/weekly_tasks.py

```diff
@@ -0,0 +1,205 @@
+    user = User(
+        username='demo_user',
+        email='demo@email.po',
+        password='s3jsd183b13',
+        full_name='The Demo'
+    )
+    return user
+
+
+def get_user(
+    demo_user: User,
+    session: Session = Depends(get_db)
+) -> User:
+    demo_username = demo_user.username
+    user = session.query(User).filter_by(username=demo_username).first()
+    if not user:
+        session.add(demo_user)
+        session.commit()
+        user_query = session.query(User)
+        user = user_query.filter_by(username=demo_username).first()
+    return user
+
+
+def get_checked_days(days: str = "") -> List[Tuple[str, str, str]]:
+    """Produces the input checked_days for the template add_edit_weekly_task"""
+    days_list = days.split(", ")
+    checked_days = []
+    for day_full_name in calendar.day_name:
+        day = day_full_name[:3]
+        day_lower = day.lower()
+        if day in days_list:
+            checked_days.append((day_full_name, day_lower, "checked"))
+        else:
+            checked_days.append((day_full_name, day_lower, ""))
+    return checked_days
+
+
+def get_days_string(
+    sun: bool, mon: bool,
+    tue: bool, wed: bool,
+    thu: bool, fri: bool,
+    sat: bool
+) -> str:
+    """Produces a string of all the days that were checked,
+    For use in the model weekly tasks."""
+    days_dict = {
+        "Mon": mon,
+        "Tue": tue,
+        "Wed": wed,
+        "Thu": thu,
+        "Fri": fri,
+        "Sat": sat,
+        "Sun": sun
+    }
+    days_list = [day for day, is_checked in days_dict.items() if is_checked]
+    days = ", ".join(days_list)
+    return days
+
+
+@router.get('/')
+async def weekly_tasks_manager(
+        request: Request,
+        session=Depends(get_db),
+        demo_user=Depends(get_placeholder_user)):
+
+    user = get_user(demo_user, session)
+
+    # TODO: Move the below function to a compatible location
+    # Need to run regularly whenever there are tasks on the week
+    # Or will run on the background after the user left the
+    # weekly-tasks manager page
+    # function:
+    # generate_tasks(user, session)  # imported from app.internal.weekly_tasks
+    # session.close()
+
+    return templates.TemplateResponse("weekly_tasks_manager.html", {
+        "request": request,
+        "weekly_tasks": user.weekly_tasks
+    })
+
+
+@router.get('/add')
+def add_weekly_task(request: Request):
+
+    checked_days = get_checked_days()
+    return templates.TemplateResponse("add_edit_weekly_task.html", {
+        "request": request,
+        "weekly_task": None,
+        "mode": "add",
+        "checked_days": checked_days
+    })
+
+
+@router.delete('/')
+def delete_weekly_task(
+        remove_id: int,
+        session=Depends(get_db)):
+
+    remove_weekly_task(remove_id, session)
+    session.close()
+    url = router.url_path_for("weekly_tasks_manager")
+    return RedirectResponse(url=url, status_code=HTTP_303_SEE_OTHER)
+
+
+@router.post('/edit')
```

Comment:
This function doesn't really edit the data in the database. Should it really be `post`?